### PR TITLE
dumpling(dm): move dumpling/utils.go to pkg/dumpling/utils.go

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve? 

as title. 

Due to https://github.com/pingcap/ticdc/pull/3894 will use `ParseExtraArgs`, move it to pkg/dumpling and rename `parseExtraArgs` to `ParseExtraArgs`.

### What is changed and how it works?

- no change

### Check List 

Tests 

 - Unit test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by lichunzhu — APPROVED** [review]

File: dm/pkg/dumpling/utils.go [hunk 1]
```diff
@@ -200,3 +205,114 @@ func ParseFileSize(fileSizeStr string, defaultSize uint64) (uint64, error) {
...
+		dumpCfg.TableFilter = ff // overwrite `block-allow-list`.
+		logger.Warn("overwrite `block-allow-list` by `tables-list` or `filter`")
+	}
+
+	return nil
+}
+
+// parseTableFilter parses `--tables-list` and `--filter`.
+// copy (and update) from https://github.com/pingcap/dumpling/blob/6f74c686e54183db7b869775af1c32df46462a6a/cmd/dumpling/main.go#L222.
+func parseTableFilter(tablesList, filters []string) (tfilter.Filter, error) {
```
Comment (Ehco1996) [inline dm/pkg/dumpling/utils.go:298]: seems we can `ParseTableFilter` in  `tidb/dumpling/export/config.go`,  so i perfer to delete this func

**Review by Ehco1996 — COMMENTED** [review]
rest LGTM

File: dm/pkg/dumpling/utils.go (hunk 1 above)
Comment (okJiang) [reply dm/pkg/dumpling/utils.go:298]: lgtm

**Review by Ehco1996 — APPROVED** [review]

**Comment by okJiang** [comment]
Because push new commit https://github.com/pingcap/ticdc/pull/3911/commits/e28c322d77f3c0bf0766e2e6be7be13368bbbc47 , so you should merge again. @Ehco1996

**Comment by Ehco1996** [comment]
@okJiang 

test Dumpling failed  please fix this  https://ci.pingcap.net/blue/organizations/jenkins/atom-ut/detail/atom-ut/5487/tests/

**Comment by okJiang** [comment]
> @okJiang
> 
> test Dumpling failed please fix this [ci.pingcap.net/blue/organizations/jenkins/atom-ut/detail/atom-ut/5487/tests](https://ci.pingcap.net/blue/organizations/jenkins/atom-ut/detail/atom-ut/5487/tests/)

It is because we used the updated `parseTableFilter` which has some changed.

I will fix it right away.

**Pull request merged**